src/helper/parse/parseBalanceQuery.ts

```diff
@@ -32,11 +34,23 @@ export function parseBalanceQuery(
     filterQuery["datetime"] = parseDateQuery(start_date, end_date);
   }
 
+  const meta: IAnyObject = {};
   for (const [key, value] of Object.entries(extra)) {
     if (isPrototypeAttribute(key)) continue;
     if (!filterQuery.meta) filterQuery.meta = {};
     filterQuery.meta[key] = value;
+
+    let newValue = value;
+    if (typeof value === "string" && isTransactionObjectIdKey(key)) {
+      newValue = new Types.ObjectId(value);
+    }
+
+    if (isValidTransactionKey(key)) {
+      filterQuery[key] = newValue;
+    } else {
+      meta[key] = newValue;
```

Comment:
I find it confusing and misleading to put both the `meta: { prop: new Value }` and `"meta.prop": newValue`.

Unit test must test for either of the two.

```js
    expect(result1).to.deep.equal({
      book: "MyBook",
      meta: { clientId, bookmarked },
    });
```

or

```js
    expect(result1).to.deep.equal({
      book: "MyBook",
      "meta.bookmarked": bookmarked,
      "meta.clientId": clientId,
    });
```

Would you please choose one of the above and implement whatever you think is the best?

Reply:
It has to be dot notation

Reply:
I cant choose one, we need to have both, please check this case:
```
const result2 = parseBalanceQuery({ clientId, bookmarked }, { name: "MyBook" });
    expect(result2).to.deep.equal({
      book: "MyBook",
      clientId,
      "meta.bookmarked": bookmarked,
      meta: { clientId, bookmarked },
    });
 ```
here `meta` is only used for snapshot balance,  the other attributes are used in the aggregation query (this is the case with custom transaction schema). The alternative is to do a bigger refactor and try to dynamically match the balance schema with the custom Transaction schema (but not sure if can be done easily). 

If you have an alternative let me know.

Reply:
cc @Uzlopak @koresar

Reply:
It seems odd. I will have a deeper look at it.